addons/project_sale_expense/models/project_project.py

```diff
@@ -16,22 +17,30 @@ def _get_expenses_profitability_items(self, with_action=True):
         query = self.env['hr.expense']._search([('is_refused', '=', False), ('state', 'in', ['approved', 'done'])])
         query.add_where('hr_expense.analytic_distribution ? %s', [str(self.analytic_account_id.id)])
         query.order = None
-        query_string, query_param = query.select('sale_order_id', 'product_id', 'array_agg(id) as ids', 'SUM(untaxed_amount) as untaxed_amount')
-        query_string = f"{query_string} GROUP BY sale_order_id, product_id"
+        query_string, query_param = query.select('sale_order_id', 'product_id', 'currency_id', 'array_agg(id) as ids', 'SUM(untaxed_amount) as untaxed_amount')
+        query_string = f"{query_string} GROUP BY sale_order_id, product_id, currency_id"
         self._cr.execute(query_string, query_param)
         expenses_read_group = [expense for expense in self._cr.dictfetchall()]
         if not expenses_read_group:
             return {}
         expenses_per_so_id = {}
         expense_ids = []
         amount_billed = 0.0
+        dict_amount_per_currency = defaultdict(lambda: 0.0)
         for res in expenses_read_group:
             so_id = res['sale_order_id']
             product_id = res['product_id']
             expenses_per_so_id.setdefault(so_id, {})[product_id] = res['ids']
             if can_see_expense:
                 expense_ids.extend(res['ids'])
-            amount_billed += res['untaxed_amount']
+            dict_amount_per_currency[res['currency_id']] += res['untaxed_amount']
+        for currency_id in dict_amount_per_currency:
+            if currency_id == self.company_id.currency_id.id:
+                amount_billed += dict_amount_per_currency[currency_id]
+                continue
+            currency = self.env['res.currency'].browse(currency_id)
+            amount_billed += currency._convert(dict_amount_per_currency[currency_id], self.company_id.currency_id, self.company_id, fields.Date.today())
```

Comment:
`fields.Date.today()` could be put in a variable to be sure we have the same value in the loop

side note: `fields.Date.today()` does not use the user timezone I think, the version in 17.0 uses `fields.Date.context_today(self)` that uses it so if there is no intended reason to use `today` we should probably use `context_today` instead